pkg/stores/partition/listprocessor/processor.go

```diff
@@ -360,3 +394,28 @@ func PaginateList(list []unstructured.Unstructured, p Pagination) ([]unstructure
 	}
 	return list[offset : offset+p.pageSize], pages
 }
+
+func FilterByProjectsAndNamespaces(list []unstructured.Unstructured, projectsOrNamespaces ProjectsOrNamespacesFilter, namespaceCache corecontrollers.NamespaceCache) []unstructured.Unstructured {
+	if len(projectsOrNamespaces.filter) == 0 {
+		return list
+	}
+	result := []unstructured.Unstructured{}
+	for _, obj := range list {
+		namespaceName := obj.GetNamespace()
+		namespace, _ := namespaceCache.Get(namespaceName)
+		if namespace == nil {
+			continue
+		}
+		projectLabel, _ := namespace.GetLabels()[projectIDFieldLabel]
```

Comment:
IIRC the annotation is the canonical source of project/namespace membership, and we should use that rather than the label where possible (and since we are just checking the string here and not using a selector, this case seems suitable more for the annotation).

Reply:
The annotation may be the source of truth but it doesn't make the label wrong. The label is kept in sync with the annotation. Using the label avoids having to do the ugly string split.

Reply:
That's fair - sounds good.